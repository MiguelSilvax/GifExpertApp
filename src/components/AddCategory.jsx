import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [text, setText] = useState("")

    const onChange = ({ target }) => {
        setText(target.value);
    }

    const onFormSubmit = ( event ) => {
        event.preventDefault();
        if (text === '') return false;
        onNewCategory(text);
        setText('');
    };


    return (

        <form
            onSubmit={ e  => onFormSubmit(e) }
            aria-label='form'
        >
            <input
                id='myTextField'
                type="text"
                placeholder="Buscar gifs"
                value={text}
                onChange={onChange}
            />
        </form>

    );
} 

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}