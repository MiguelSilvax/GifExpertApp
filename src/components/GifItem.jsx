import React from 'react';

import PropTypes from 'prop-types';


export const GifItem = ({ id , title , url }) => {
    return (
        <div key={id} className="card">
            <p aria-label='title-label'>{title}</p>
            <img src={url} alt={title} id='img'/>
        </div>
    );
      
}

GifItem.propTypes = {
    id: PropTypes.string,
    title: PropTypes.string,
    url : PropTypes.string.isRequired,
}
