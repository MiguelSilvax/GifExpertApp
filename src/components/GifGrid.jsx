import { GifItem } from "../components";
import { useFechItem } from '../../Hooks/useFechItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {
    const { images , isLoading } = useFechItem(category);
    return (
        <>
            {
                isLoading && (<h2>Cargando...</h2>)
            }
            <div className="card-grid">
                {
                    images.map((item) => {
                        return (
                            <GifItem
                                key={item.id}
                                {...item}
                            />
                        )
                    })
                }
            </div>
        </>
    );
};


GifGrid.propTypes = {
    category : PropTypes.string.isRequired,
}
