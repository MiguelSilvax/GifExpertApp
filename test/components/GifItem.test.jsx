import { getByText, render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";
describe('Pruebas para GifItem', () => {
  test('Debe coincidir con el snapshot', () => {
    const { container } = render(<GifItem id="" title="" url="" />);
    screen.debug();
    expect(container).toMatchSnapshot();
  });


  test('Debe retornar los props sus respectivos tipos de datos', () => {
    render(<GifItem id="1" title="Naruto shippuden" url="https://home/src/public/naruto.gif" />);
    expect(screen.getByRole('img')).toBeTruthy();
    screen.debug();
    console.log(screen.getByRole('img').src); // otra forma de hacer selecci√≥n a un elemento
    console.log(document.getElementById('img').src);
    console.log(document.getElementById('img').alt);
    expect(document.getElementById('img').src).toEqual( expect.any(String));
    expect(document.getElementById('img').alt).toEqual( expect.any(String));
  });
});