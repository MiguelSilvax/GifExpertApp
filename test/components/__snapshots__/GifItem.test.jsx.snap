// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para GifItem Debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p
      aria-label="title-label"
    />
    <img
      alt=""
      id="img"
      src=""
    />
  </div>
</div>
`;
